### PR TITLE
DRAFT: ET Lint and Jest ES6 update

### PR DESCRIPTION
**Pull request by ScottGarrison**
### 🎫 Ticket

[TICKET_ID] None - TBD

### 🗒️ Description
Updated Event Tickets plugin to Lint and Jest for ES6. 
Going forward all code can be written in ES6 format.

**NOTE**
This is just a Draft until `parksosaurus` is updated properly for merging

### ✔️ Checklist
- [x ] I've included a changelog entry in the readme.txt file. 
- [ ✔️] My code is tested. 
- [✔️ ] My code has [proper inline documentation](https://the-events-calendar.github.io/products-engineering/docs/code-standards/).

### CODE REVIEW

File: src/modules/blocks/tickets/template.js [hunk 1]
```diff
@@ -38,38 +42,41 @@ class Tickets extends PureComponent {
...
+			canCreateTickets,
+		} = this.props;
 
 		return (
 			<Fragment>
-				<TicketsContainer isSelected={isSelected}/>
-				{canCreateTickets && <TicketsDashboard isSelected={isSelected} clientId={clientId}/>}
-				<TicketControls/>
+				<TicketsContainer isSelected={ isSelected } />
+				{ canCreateTickets && <TicketsDashboard isSelected={ isSelected } clientId={ clientId } />}
```
Comment (codingmusician) [inline src/modules/blocks/tickets/template.js:53]: We should add a space before the ending bracket, too.
```suggestion
				{ canCreateTickets && <TicketsDashboard isSelected={ isSelected } clientId={ clientId } /> }
```

File: src/modules/blocks/tickets/template.js [hunk 2]
```diff
@@ -38,38 +42,41 @@ class Tickets extends PureComponent {
...
-						{__( 'Read about our plans for future features.', 'event-tickets' )}
+						{ __( 'Read about our plans for future features.', 'event-tickets' ) }
 					</a>
-					<br/>
-					<a href="#" onClick={() => wp.data.dispatch('core/block-editor').removeBlock(clientId)}>{__( 'Remove block', 'event-tickets' )}</a>
+					<br />
+					<Button variant="secondary" onClick={ () =>
+						wp.data.dispatch( 'core/block-editor' ).removeBlock( clientId )
+					}>
+						{__( 'Remove block', 'event-tickets' )}
```
Comment (codingmusician) [inline src/modules/blocks/tickets/template.js:78]: Spacing.
```suggestion
						{ __( 'Remove block', 'event-tickets' ) }
```

File: src/resources/js/v2/tickets-commerce.js [hunk 3]
```diff
@@ -200,7 +184,7 @@ tribe.tickets.commerce = {};
 			return purchaser;
 		}
 
-		purchaser.name  = $container.find( obj.selectors.purchaserName ).val();
+		purchaser.name = $container.find( obj.selectors.purchaserName ).val();
 		purchaser.email = $container.find( obj.selectors.purchaserEmail ).val();
```
Comment (codingmusician) [inline src/resources/js/v2/tickets-commerce.js:188]: In this case, we're aligning the equals signs with the `purchaser.email` line just so it's easier to read.

Comment (ScottGarrison) [reply src/resources/js/v2/tickets-commerce.js:188]: Sure, but it wasn't a change I just made randomly. It was noted by eslint for extra spaces.

Comment (codingmusician) [reply src/resources/js/v2/tickets-commerce.js:188]: Oh... In that case, never mind. LOL

**Pull request closed without merging**